easybuild/tools/toolchain/toolchain.py

```diff
@@ -108,6 +116,13 @@ def __init__(self, name=None, version=None, mns=None):
                 self.mod_short_name = self.mns.det_short_module_name(tc_dict)
                 self.init_modpaths = self.mns.det_init_modulepaths(tc_dict)
 
+    def add_class_constants_to_restore(self, names):
+        """Add given constants to list of class constants to restore with each new instance."""
+        if self.CLASS_CONSTANTS_TO_RESTORE is None:
```

Comment:
i don't understand the benefit of None. initial `CLASS_CONSTANTS_TO_RESTORE` could be `[]`, and you wouldn't need a method at all.